src/decoder/error.rs

```diff
@@ -98,17 +113,10 @@ impl error::Error for DecoderError {
                 inner.description()
             }
             DecoderError::UnrecognizedDocumentElementType { .. } => "unrecognized element type",
-            DecoderError::InvalidArrayKey(..) => "invalid array key",
```

Comment:
Should we take this time to remove the explicit `description` implementations on our errors given that it's deprecated (and has a default implementation)?

Reply:
done. Doing this also made me realize I forgot to document / label the `oid` error cases, so I've done that as well.